### ISSUE TITLE
Feature ideas

### ISSUE THREAD

**Issue by jedateach** [jedateach/OpenSeadragonAreaSelector#1]
## Feature / Architecture Planning

This plugin should provide equivelant functionality to jQuery UI's `draggable` and `resizeable` interactions. It may even be worth considering using jQuery UI to do the heavy lifting.
### MVP Features
- ~~Create + attach an area selector to a viewer~~.
- ~~Init location + size~~
- ~~Click/press and drag to move the area selector~~.
- ~~Click/press and drag handles (corners / edges) to resize.~~
- ~~Constrain position+size to the maximum bounds of image, or possibly a user-defined area.~~
### Backlog

Most of these ideas are from jQuery UI:
- ~~Snap to grid~~
- Event handling: provide hooks to do things. 
- Enabled / disable specific options
- Cursor style
- Auto scroll: pan the viewer as the selector overlaps the edge of the viewer.
- Rotation
- Visual feedback
- Animate resizing
- Resize helper: Display only an outline of the element while resizing

**Comment by sharmaashish** [comment]
Nice plugin. any thoughts on how a user might interact with this before an event is triggered? Say something like the following:
User wants to zoom-in on a ROI --> Uses your plugin to adjust the ROI --> Detection Completion (??) and then zoom-in.
Though alternatively, you could just provide getter's to allow external actions to fetch the extents of the BOX and do as they please. 

btw, the Resize Helper would be a nice feature

**Comment by jedateach** [comment]
@sharmaashish Yes that is certainly one use case. You could also use it to define overlays that you add to the image permanently. Perhaps people could comment / annotate areas they select.
Another idea I've had is you could define multiple areas, and they then become a sequenced "tour" around the image.

As always, pull requests are most welcome!
